osf-cli/pom.xml

```diff
@@ -61,34 +61,39 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-assembly-plugin</artifactId>
-                <configuration>
-                    <appendAssemblyId>false</appendAssemblyId>
-                    <attach>true</attach>
-                </configuration>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>2.4.3</version>
                 <executions>
                     <execution>
-                        <id>standalone-jar</id>
                         <phase>package</phase>
                         <goals>
-                            <goal>single</goal>
+                            <goal>shade</goal>
                         </goals>
                         <configuration>
-                            <archive>
-                                <manifest>
+                            <transformers>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
                                     <mainClass>org.dataconservancy.cos.packaging.cli.PackageGenerationApp</mainClass>
-                                    <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
-                                    <addDefaultSpecificationEntries>true</addDefaultSpecificationEntries>
-                                </manifest>
-                                <manifestEntries>
-                                    <Implementation-Title>${project.name}</Implementation-Title>
-                                    <Implementation-Version>${project.version} (${buildNumber})</Implementation-Version>
-                                </manifestEntries>
-                            </archive>
-                            <finalName>${pkg.gen.assembly.standalone.fileName}</finalName>
-                            <descriptorRefs>
-                                <descriptorRef>jar-with-dependencies</descriptorRef>
-                            </descriptorRefs>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
+                                    <resource>META-INF/spring.handlers</resource>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
+                                    <resource>META-INF/spring.schemas</resource>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
+                                    <resource>META-INF/services/org.apache.jena.system.JenaSubsystemLifecycle</resource>
+                                </transformer>
+                            </transformers>
+                            <filters>
```

Comment:
This has to do with getting a Security Exception when building the uber 
jar with the shade plugin. This excludes the signature files, which 
bypasses the check I think. For some reason, it is the shading which 
causes the issue.

On 08/02/2016 03:29 PM, Elliot Metsger wrote:
>
> In osf-cli/pom.xml 
> <https://github.com/emetsger/osf-packaging/pull/6#discussion_r73221098>:
>
> > -                            <finalName>${pkg.gen.assembly.standalone.fileName}</finalName>
> > -                            <descriptorRefs>
> > -                                <descriptorRef>jar-with-dependencies</descriptorRef>
> > -                            </descriptorRefs>
> > +                                </transformer>
> > +                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
> > +                                    <resource>META-INF/spring.handlers</resource>
> > +                                </transformer>
> > +                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
> > +                                    <resource>META-INF/spring.schemas</resource>
> > +                                </transformer>
> > +                                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
> > +                                    <resource>META-INF/services/org.apache.jena.system.JenaSubsystemLifecycle</resource>
> > +                                </transformer>
> > +                            </transformers>
> > +                            <filters>
>
> Out of curiosity, were these files (META-INF/*.SF and friends) causing 
> problems? Is that why they are filtered out?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/emetsger/osf-packaging/pull/6/files/ebccbed8c1362bb6545e2fd644f9e5f50e3bf929#r73221098>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAS-t-LpZLPzw03yoviw5ifEupR8Sz7Cks5qb5p8gaJpZM4JXnbx>.
>